src/futils.c

```diff
@@ -21,32 +22,30 @@ int git_futils_mkpath2file(const char *file_path, const mode_t mode)
 		GIT_MKDIR_PATH | GIT_MKDIR_SKIP_LAST | GIT_MKDIR_VERIFY_DIR);
 }
 
+/* Reimplementation of mkstemp, except with a umask-modified mode, instead of 0600. */
 int git_futils_mktmp(git_buf *path_out, const char *filename, mode_t mode)
 {
+	/* glibc tries 238328 times, OpenBSD tries INT_MAX times. */
+	unsigned int tries = INT_MAX;
 	int fd;
-	mode_t mask;
+	const int open_flags = O_RDWR | O_CREAT | O_EXCL | O_BINARY | O_CLOEXEC;
 
-	p_umask(mask = p_umask(0));
+	while (tries--) {
+		git_buf_sets(path_out, filename);
+		git_buf_puts(path_out, "_git2_");
+		git_buf_put_rand(path_out, 8);
```

Comment:
I'd argue that manually generating the name doesn't make a lot of sense. Instead, we should probably be using something like tempnam(3P) that properly handles this for us, and that would also remove the need for `git_buf_put_rand` and the random-seeding hassle.